lib/stanford-mods/searchworks.rb

```diff
@@ -389,46 +389,38 @@ def place
       # If no such dates, select the first date in the dates_marc_encoding array.  Otherwise return nil
       # @return [String] value for the pub_date_display Solr field for this document or nil if none
       def pub_date_display
-          return dates_no_marc_encoding.first unless dates_no_marc_encoding.empty?
-          return dates_marc_encoding.first    unless dates_marc_encoding.empty?
-          return nil
+        return dates_no_marc_encoding.first unless dates_no_marc_encoding.empty?
+        return dates_marc_encoding.first unless dates_marc_encoding.empty?
+        nil
       end
 
       # For the date indexing, sorting and faceting, the first place to look is in the dates with encoding=marc array.
       # If that doesn't exist, look in the dates without encoding=marc array.  Otherwise return nil
       # @return [Array<String>] values for the date Solr field for this document or nil if none
       def pub_dates
-        return dates_marc_encoding    unless dates_marc_encoding.empty?
+        return dates_marc_encoding unless dates_marc_encoding.empty?
         return dates_no_marc_encoding unless dates_no_marc_encoding.empty?
-        return nil
-      end
-
-      def is_number?(object)
-        true if Integer(object) rescue false
-      end
-      def is_date?(object)
-        true if Date.parse(object) rescue false
+        nil
       end
 
       # Get the publish year from mods
       # @return [String] 4 character year or nil if no valid date was found
       def pub_year
-        #use the cached year if there is one
+        # use the cached year if there is one
         if @pub_year
-          if @pub_year == ''
-            return nil
-          end
+          return nil if @pub_year == ''
           return @pub_year
         end
+
         dates = pub_dates
         if dates
           pruned_dates = []
           dates.each do |f_date|
-            #remove ? and []
-            if (f_date.length == 4 && f_date.end_with?('?'))
-              pruned_dates << f_date.gsub('?','0')
+            # remove ? and []
+            if f_date.length == 4 && f_date.end_with?('?')
+              pruned_dates << f_date.tr('?', '0')
             else
-              pruned_dates << f_date.gsub('?','').gsub('[','').gsub(']','')
+              pruned_dates << f_date.tr('?', '').tr('[', '').tr(']', '')
```

Comment:
`#tr(x, '')` could also be written `#delete(x)`, and you could do e.g. `.delete('?[]')` to remove them all in one go.